### DISCUSSION TITLE
No way to select an item in index when `authorizedToDelete() = false` and you have `sole()` actions.

### DISCUSSION THREAD

**Discussion by ekateiva** [laravel/nova-issues#6255] [Ideas & Feature Requests]
- Laravel Version: 10.18.0
- Nova Version: v4.26.4 (Silver Surfer)
- PHP Version: 8.2.14
- Browser type and version: Chrome 120.0.6099.216

### Description:
I want to have `sole()` actions that I expect I could select from index. However, I also want the resource to be not _deletable_. When we have this scenario, there is no select checkboxes for the resources in the inde.

<img width="573" alt="Screenshot 2024-01-19 at 15 28 38" src="https://github.com/laravel/nova-issues/assets/12799259/a9399a2e-9ffa-447a-aa82-db7301ae0b6e">

### Detailed steps to reproduce the issue on a fresh Nova installation:
- In the resource, make an item not available to delete by `authorizedToDelete()` returning `false`.
- Create an action for the resource as `sole()` action.

E.g.:
```php
...

public function authorizedToDelete(Request $request): bool
{
    return false;
}

public function actions(NovaRequest $request): array
{
    return [
        SomeSoleAction::make()
            ->sole(),
    ];
}
...
```

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

**Comment by ekateiva** [comment]
> Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

@crynobone 
Please, see the repository, where I reproduce the issue: https://github.com/ekateiva/laravel-nova-issue-6157
I added there in `README` the information how to reproduce the issue.

Basically, this is the code that reproduces the issue:
https://github.com/ekateiva/laravel-nova-issue-6157/blob/main/app/Nova/OrderItem.php#L56-L74

Let me know if you need any more info, but it should be clear now.

**Comment by crynobone** [comment]
This is something that can't easily be fixed without reducing the performance of Index page and I would suggest displaying "Sole" action to show as inline for now: https://nova.laravel.com/docs/actions/registering-actions.html#inline-actions

**Reply by ekateiva** [reply]
Unfortunately, _inline action_ is also not an option to go if we are speaking about the performance. 
If our action has `fields()` that depend on selected model, it will be performing all requests to the database (and so on) on index for all rows and not only for the selected one (as this is how _inline actions_ works in Nova now ). 😞
